frontend/src/app/offers/offers.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { Headers, Http, Response, RequestOptions } from '@angular/http';
+import { Observable } from 'rxjs/Rx';
+
+@Injectable()
+export class OffersService {
+  private url = 'http://localhost:8000/api/offers/';
+
+  constructor (
+    private http: Http
+  ) {}
+
+  getOffers() {
+    let headers: Headers = new Headers({
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    });
+    let options: RequestOptions = new RequestOptions({ headers: headers });
+
+    return this.http.get(this.url, options)
+      .map((res:Response) => res.json())
+      .catch(this.handleError);
+  }
+
+  handleError(reject: Response | any) {
```

Comment:
Don't see a reason fot this method existence

Reply:
Shouldn't we somehow handle API errors?